apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/utility/DataUtilityCommons.java

```diff
@@ -0,0 +1,184 @@
+package gov.cms.bfd.data.fda.utility;
+
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CharsetEncoder;
+import java.nio.charset.CodingErrorAction;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Provides an FDA Drug Code file. */
+public class DataUtilityCommons {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataUtilityCommons.class);
+
+  /** Size of the buffer to read/write data. */
+  private static final int BUFFER_SIZE = 4096;
+
+  /**
+   * Gets the fda drug codes from the fda file.
+   *
+   * @param outputDir the output directory.
+   * @param fdaFile the fda file.
+   */
+  public static void getFDADrugCodes(String outputDir, String fdaFile) {
```

Comment:
probably need to add a "throws IllegalStateException" to this method.

Reply:
Aslo, where you add the throws on all your methods, make sure the javadoc represents the changes.